### PR TITLE
[DOC] Refresh standard libraries page

### PR DESCRIPTION
**Pull request by st0012**
I think the [standard libraries page](https://docs.ruby-lang.org/en/master/standard_library_rdoc.html) could use a little love so I:

- Migrated its markup to markdown
- Improved the libraries listing and linked to the source code
- Fixed some grammar and improved the phrasing

Result:

https://github.com/user-attachments/assets/f00caeea-ccc3-483e-94cb-a8fa337951d2

The descriptions and some libraries' placements may use some more detailed reviews, but I plan to do it in separate PRs.

### CODE REVIEW

File: doc/standard_library.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+- Default gems are not uninstallable from the Ruby installation.
+- Default gems can be updated using rubygems.
+    - e.g. `gem update json`
+- Default gems can be used with bundler environments like `unbundled_env`.
+- Default gems can be used at any version in a Gemfile.
+    - e.g. `gem "json", ">= 2.6"`
+
+### Libraries
+
+- [benchmark][benchmark]: Provides methods to measure and report the time used to execute code
```
Comment (nobu) [inline doc/standard_library.md:36]: Why these lines double the library names?

File: doc/standard_library.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
+# Ruby Standard Library
+
+The Ruby Standard Library is a large collection of classes and modules you can
+require in your code to gain additional features.
+
+Below is an overview of the libraries and extensions, followed by a brief description
+of each.
+
+## Libraries
```
Comment (nobu) [inline doc/standard_library.md:9]: Shouldn't these headings `###` as well as under Default/Bundled gems?

Comment (st0012) [reply doc/standard_library.md:9]: I misread the original headings. I've restored them now.

File: doc/standard_library.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+- `MakeMakefile`: A module used to generate a Makefile for C extensions
+- `RbConfig`: Information about your Ruby configuration and build
+- `Gem`: A package management framework for Ruby
+
+## Extensions
+
+- `Coverage`: Provides coverage measurement for Ruby.
+- `Monitor`: Provides an object or module that can be safely used by more than one thread.
```
Comment (jeremyevans) [inline doc/standard_library.md:18]:
```suggestion
- `Monitor`: Provides a reentrant mutex.
```

File: doc/standard_library.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+- [tsort]: Topological sorting using Tarjan's algorithm
+- [un]: Utilities to replace common UNIX commands
+- [uri]: A Ruby module providing support for Uniform Resource Identifiers
+- [yaml]: The Ruby client library for the Psych YAML implementation
+- [weakref]: Allows a referenced object to be garbage-collected
+
+## Extensions
+
+- [date]: A subclass of Object that includes the Comparable module for handling dates
+- `DateTime`: A subclass of Date for handling dates, hours, minutes, seconds, and offsets
```
Comment (jeremyevans) [inline doc/standard_library.md:80]:
```suggestion
- [date]: Represents dates, with a subclass for dates with time and timezones.
```

I realize `DateTime` was listed separately previously, but I think that was a mistake.  We should only be listing the library here (`DateTime` is not a separate library, just a class provided by the `date` library).

File: doc/standard_library.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+# Bundled gems
+
+- Bundled gems are shipped with Ruby releases and also available as rubygems.
+    - They are only bundled with Ruby releases.
+    - They can be uninstalled from the Ruby installation.
+    - They need to be declared in a Gemfile when used with bundler.
+
+## Libraries
+
+- [minitest]: A test suite with TDD, BDD, mocking, and benchmarking
```
Comment (jeremyevans) [inline doc/standard_library.md:104]:
```suggestion
- [minitest]: A test library supporting TDD, BDD, mocking, and benchmarking
```

**Review by jeremyevans — CHANGES_REQUESTED** [review]
I think we should continue linking default gems to the main class of the library (as currently done on https://docs.ruby-lang.org/en/master/standard_library_rdoc.html), not using website links to GitHub repositories.  It's way more friendly for users.  Using web links for bundled gems is fine, as we aren't generating documentation for those, and the web link is better than the current lack of link.

**Comment by st0012** [comment]
> I think we should continue linking default gems to the main class of the library

That's a good call. I've restored all the autolinks but IMO it's still helpful to list their GitHub links too. Otherwise it's not easy to find that information from the generated docs either.

File: doc/standard_library.md [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+
+- Bundled gems are shipped with Ruby releases and also available as rubygems.
+    - They are only bundled with Ruby releases.
+    - They can be uninstalled from the Ruby installation.
+    - They need to be declared in a Gemfile when used with bundler.
+
+## Libraries
+
+- [minitest]: A test library supporting TDD, BDD, mocking, and benchmarking
+- [power_assert]: Power Assert for Ruby.
```
Comment (jeremyevans) [inline doc/standard_library.md:107]:
```suggestion
- [power_assert]: Power Assert for Ruby
```

File: doc/standard_library.md [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+- [net-ftp]: Support for the File Transfer Protocol
+- [net-imap]: Ruby client API for the Internet Message Access Protocol
+- [net-pop]: Ruby client library for POP3
+- [net-smtp]: Simple Mail Transfer Protocol client library for Ruby
+- [matrix]: Represents a mathematical matrix.
+- [prime]: Prime numbers and factorization library
+- [rbs]: RBS is a language to describe the structure of Ruby programs
+- [typeprof]: A type analysis tool for Ruby code based on abstract interpretation
+- [debug]: Debugging functionality for Ruby
+- [racc]: A LALR(1) parser generator written in Ruby.
```
Comment (jeremyevans) [inline doc/standard_library.md:121]:
```suggestion
- [racc]: A LALR(1) parser generator written in Ruby
```

File: doc/standard_library.md [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+- [power_assert]: Power Assert for Ruby.
+- [rake]: Ruby build program with capabilities similar to make
+- [test-unit]: A compatibility layer for MiniTest
+- [rexml]: An XML toolkit for Ruby
+- [rss]: A family of libraries supporting various XML-based "feeds"
+- [net-ftp]: Support for the File Transfer Protocol
+- [net-imap]: Ruby client API for the Internet Message Access Protocol
+- [net-pop]: Ruby client library for POP3
+- [net-smtp]: Simple Mail Transfer Protocol client library for Ruby
+- [matrix]: Represents a mathematical matrix.
```
Comment (jeremyevans) [inline doc/standard_library.md:116]:
```suggestion
- [matrix]: Represents a mathematical matrix
```

File: doc/standard_library.md [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+- `Gem`: A package management framework for Ruby
+
+## Extensions
+
+- `Coverage`: Provides coverage measurement for Ruby.
+- `Monitor`: Provides a reentrant mutex.
+- `objspace`: Extends the ObjectSpace module to add methods for internal statistics.
+- `PTY`: Creates and manages pseudo-terminals.
+- `Ripper`: Provides an interface for parsing Ruby programs into S-expressions.
+- `Socket`: Accesses underlying OS socket implementations.
```
Comment (jeremyevans) [inline doc/standard_library.md:22]:
```suggestion
- `Coverage`: Provides coverage measurement for Ruby
- `Monitor`: Provides a reentrant mutex
- `objspace`: Extends the ObjectSpace module to add methods for internal statistics
- `PTY`: Creates and manages pseudo-terminals
- `Ripper`: Provides an interface for parsing Ruby programs into S-expressions
- `Socket`: Accesses underlying OS socket implementations
```

**Review by jeremyevans — APPROVED** [review]
Looks good.  Only issue I can see is that we should be consistent with not including `.` at the end of some of the bullet points.

**Pull request merged**